src/global/global_init.cc

```diff
@@ -268,17 +269,34 @@ global_init(const std::map<std::string,std::string> *defaults,
     if (g_conf()->setgroup.length() > 0) {
       gid = atoi(g_conf()->setgroup.c_str());
       if (!gid) {
-	char buf[4096];
-	struct group gr;
-	struct group *g = 0;
-	getgrnam_r(g_conf()->setgroup.c_str(), &gr, buf, sizeof(buf), &g);
-	if (!g) {
-	  cerr << "unable to look up group '" << g_conf()->setgroup << "'"
-	       << ": " << cpp_strerror(errno) << std::endl;
-	  exit(1);
+	long start_size = sysconf (_SC_GETGR_R_SIZE_MAX);
+	if (start_size < 0) {
+	  start_size = 4096;
+	}
+	static constexpr std::size_t max_size = 16 * 1024;
```

Comment:
If the limit is just 16K, why not just have an on-stack array of that size, making this PR a one-liner?

Reply:
This is a very good point, especially since we're only doing it once at startup.